dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -1653,24 +1688,33 @@ private void bindEventParamsForUpdate( PreparedStatement ps, ProgramStageInstanc
         ps.setTimestamp( 8, JdbcEventSupport.toTimestamp( new Date() ) );
         ps.setLong( 9, programStageInstance.getAttributeOptionCombo().getId() );
         ps.setString( 10, programStageInstance.getStoredBy() );
-        ps.setString( 11, programStageInstance.getCompletedBy() );
-        ps.setBoolean( 12, programStageInstance.isDeleted() );
-        ps.setString( 13, programStageInstance.getCode() );
-        ps.setTimestamp( 14, JdbcEventSupport.toTimestamp( programStageInstance.getCreatedAtClient() ) );
-        ps.setTimestamp( 15, JdbcEventSupport.toTimestamp( programStageInstance.getLastUpdatedAtClient() ) );
-        ps.setObject( 16, JdbcEventSupport.toGeometry( programStageInstance.getGeometry() ) );
+        ps.setObject( 11, userInfoToJson( programStageInstance.getLastUpdatedByUserInfo() ) );
+        ps.setString( 12, programStageInstance.getCompletedBy() );
+        ps.setBoolean( 13, programStageInstance.isDeleted() );
+        ps.setString( 14, programStageInstance.getCode() );
+        ps.setTimestamp( 15, JdbcEventSupport.toTimestamp( programStageInstance.getCreatedAtClient() ) );
+        ps.setTimestamp( 16, JdbcEventSupport.toTimestamp( programStageInstance.getLastUpdatedAtClient() ) );
+        ps.setObject( 17, JdbcEventSupport.toGeometry( programStageInstance.getGeometry() ) );
 
         if ( programStageInstance.getAssignedUser() != null )
         {
-            ps.setLong( 17, programStageInstance.getAssignedUser().getId() );
+            ps.setLong( 18, programStageInstance.getAssignedUser().getId() );
         }
         else
         {
-            ps.setObject( 17, null );
+            ps.setObject( 18, null );
         }
 
-        ps.setObject( 18, eventDataValuesToJson( programStageInstance.getEventDataValues(), this.jsonMapper ) );
-        ps.setString( 19, programStageInstance.getUid() );
+        ps.setObject( 19, eventDataValuesToJson( programStageInstance.getEventDataValues(), this.jsonMapper ) );
+        ps.setString( 20, programStageInstance.getUid() );
+    }
+
+    @SneakyThrows
+    private PGobject userInfoToJson(ProgramStageInstanceUserInfo userInfo) {
```

Comment:
Move to `EventUtils` class as static method

Reply:
done, thank you